lib/src/presentation/pages/movie_details.dart

```diff
@@ -1,43 +1,68 @@
 import 'package:flutter/material.dart';
+import '../../core/util/enums.dart';
+import '../../domain/entity/cast.dart';
+import '../../domain/entity/event.dart';
+import '../widgets/loaders/cast_loader.dart';
 import '../../core/util/ui_consts.dart';
 import '../../domain/entity/movie.dart';
+import '../bloc/movie_bloc.dart';
 import '../widgets/movie/movie_presentation.dart';
 import '../widgets/movie/overview.dart';
 import '../widgets/movie/movie_details_custom_nav_bar.dart';
-import '../widgets/movie/cast.dart';
+import '../widgets/movie/cast_widget.dart';
+import '../widgets/shared/error_class.dart';
 
-class MovieDetails extends StatefulWidget {
-  const MovieDetails({
-    super.key,
-    required this.movie,
-  });
+class MovieDetails extends StatelessWidget {
+  static const double endPadding = 130;
+  static const String message = 'Error to bring the cast';
 
   final Movie movie;
-  static const String routeName = '/movie';
-
-  @override
-  State<MovieDetails> createState() => _MovieDetailsState();
-}
+  final MovieBloc movieBloc;
 
-class _MovieDetailsState extends State<MovieDetails> {
-  static const double endPadding = 130;
+  const MovieDetails({
+    required this.movie,
+    required this.movieBloc,
+    super.key,
+  });
 
   @override
   Widget build(BuildContext context) {
+    movieBloc.setLastMovie(movie);
+    movieBloc.loadCast(movie);
```

Comment:
Try to made this call outside the build

Reply:
For that i have to do the widget Statefull. I can not find the message but i am almost sure that Mati tell me that is not bad doing it in the build i can not remeber why he told me that. Should i remove it and make the widget statefull?

Reply:
Clearly i was wrong, sorry for that!